packages/itwinui-react/src/core/Menu/MenuItem.tsx

```diff
@@ -139,112 +197,134 @@ export const MenuItem = React.forwardRef((props, forwardedRef) => {
       case 'Enter':
       case ' ':
       case 'Spacebar': {
-        !disabled && onClick?.(value);
-        event.preventDefault();
-        break;
-      }
-      case 'ArrowRight': {
-        if (subMenuItems.length > 0) {
-          setIsSubmenuVisible(true);
-
-          // flush and reset state so we are ready to focus again next time
-          flushSync(() => setFocusOnSubmenu(true));
-          setFocusOnSubmenu(false);
-
-          event.preventDefault();
-          event.stopPropagation();
-        }
-        break;
-      }
-      case 'ArrowLeft': {
-        if (parent.ref) {
-          parent.ref.current?.focus();
-          parent.setIsNestedSubmenuVisible(false);
-        }
-        event.stopPropagation();
+        onActivate();
         event.preventDefault();
         break;
       }
-      case 'Escape': {
-        // focus might get lost if submenu closes so move it back to parent
-        parent.ref?.current?.focus();
-        break;
-      }
       default:
         break;
     }
   };
 
+  const onMouseEnter = (e: React.MouseEvent<HTMLElement>) => {
+    // Focus the item when hovered.
+    if (e.target === e.currentTarget) {
+      menuItemRef.current?.focus();
+    }
+  };
+
+  const onFocus = () => {
+    parent.setHasFocusedNodeInSubmenu(true);
+
+    tree?.events.emit('onNodeFocused', {
+      nodeId,
+      parentId,
+    } satisfies TreeEvent);
+  };
+
+  const onClick = () => {
+    onActivate();
+  };
+
+  const onActivate = () => {
+    !disabled && subMenuItems.length === 0 && onClickProp?.(value);
+    setIsSubmenuVisible((prev) => !prev);
+  };
```

Comment:
1. Since `onActivate` and now identical, they can be combined.
2. We should call `onClickProp` even when `subMenuItems` is provided, in case users were relying on that behavior.
3. We should not call `setIsSubmenuVisible` on disabled items because #1925`.
   - Additionally, we might also want to disable all other triggers on disabled items to make it consistent.

Reply:
1. Combined (https://github.com/iTwin/iTwinUI/pull/1942/commits/a9ab28f140a4e8d0bb3d63e08735697a093f900e).
2. Ohh, sounds good, updated (https://github.com/iTwin/iTwinUI/pull/1942/commits/72058bb581bccdfc6867fc3ec4b0190faa5cd193).
3. I thought we planned to handle disabled `MenuItem`s later, but yes, reading your comment on the issue you linked, it feels better to handle it here. Handled (https://github.com/iTwin/iTwinUI/pull/1942/files/72058bb581bccdfc6867fc3ec4b0190faa5cd193..c556efa6d4ab39f823f80930084462a98303b7e7). These commits also disable all Floating UI triggers/interactions and our event handlers for disabled items.

Reply:
Why do you need to disable everything in so many places? We don't need all this extra code. It would be enough just _not_ render the submenu when the triggering menu-item is disabled.

Reply:
> It would be enough just _not_ render the submenu when the triggering menu-item is disabled.

I did it this way since I thought we wanted to keep `isSubmenuVisible` as the source of truth (https://github.com/iTwin/iTwinUI/pull/1942#discussion_r1567940451). But I guess that discussion was only for non-disabled `MenuItem`s.

So, I replaced that code with the safer approach of not rendering the submenu when the triggering `MenuItem` is disabled (https://github.com/iTwin/iTwinUI/pull/1942/commits/6e39132e65d42d9f6c972c0d729ecb755a7067b3).